src/server/set_family.cc

```diff
@@ -1378,6 +1381,36 @@ void SInterStore(CmdArgList args, ConnectionContext* cntx) {
   (*cntx)->SendLong(result->size());
 }
 
+void SInterCard(CmdArgList args, ConnectionContext* cntx) {
+  unsigned num_keys;
+  if (!absl::SimpleAtoi(ArgS(args, 0), &num_keys)) {
+    return (*cntx)->SendError("numkeys should be greater than 0");
+  } else if (args.size() == (num_keys + 3) && num_keys > args.size() - 3) {
+    return (*cntx)->SendError("Number of keys can't be greater than number of args");
+  }
```

Comment:
1. absl::SimpleAtoi returns false if it failed to parse the integer for any reason, so for `SINTECARD NotAnInt` it'll return the error `numkeys should be greater than 0`. I suppose Redis just reports the basic invalid integer error and we can do the same
2. The first condition is impossible 😀, you simply check `a == b + 3 && b > a - 3` which is `a == b + 3 && b + 3 > a`. By the way, I'd check this condition only after you determined whether LIMIT is present so you don't have to do it twice and the code becomes much simpler